src/App.tsx

```diff
@@ -1,204 +1,188 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import { UserWarning } from './UserWarning';
 import { USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { TodoList } from './components/todoList';
+import { TodoForm } from './components/todoForm';
+import { Filter } from './components/filter';
+import { Notification } from './components/notification';
+
+enum FilterType {
+  FILTERED_ACTIVE = 'Active',
+  FILTERED_COMPLETED = 'Completed',
+  CLEAR_COMPLETED = 'Clear completed',
+  UNFILTERED = 'All',
+}
 
 export const App: React.FC = () => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [titleInput, setTitleInput] = useState('');
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [loading, setLoading] = useState(true);
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [hidden, setHidden] = useState(true);
+
+  const [filterBy, setFilterBy] = useState(FilterType.UNFILTERED);
+
+  const todoCount = todos.length;
+
+  function clear(todoId: number) {
+    todoService.deleteTodos(todoId).then(() => {
+      setTodos(currentTodos => currentTodos.filter(todo => todo.id !== todoId));
+    });
+  }
+
+  function getPreparedTodos(todoList: Todo[], filterType: string) {
+    const preparedTodos = [...todoList];
+
+    if (filterType === FilterType.UNFILTERED) {
+      return preparedTodos;
+    }
+
+    if (filterType === FilterType.FILTERED_COMPLETED) {
+      return preparedTodos.filter(todo => todo.completed);
+    }
+
+    if (filterType === FilterType.FILTERED_ACTIVE) {
+      return preparedTodos.filter(todo => !todo.completed);
+    } else {
+      return null;
```

Comment:
default case - just return current todos as well

Or throw a new error (in fact it should never happen)